### PR TITLE
[numpy] numpy copysign

### PR DESCRIPTION
**Pull request by hzfan**
## Description ##
Numpy compatible [copysign](https://docs.scipy.org/doc/numpy/reference/generated/numpy.copysign.html)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- add copysign, including doc and test

## Comments ##
- Thank @haojin2 and others for review.

### CODE REVIEW

**Comment by hzfan** [comment]
@mxnet-label-bot add [Numpy]

File: src/operator/numpy/np_elemwise_broadcast_op.cc [hunk 1]
```diff
@@ -198,5 +218,25 @@ MXNET_OPERATOR_REGISTER_NP_BINARY_SCALAR(_npi_minimum_scalar)
...
+
+MXNET_OPERATOR_REGISTER_NP_BINARY_SCALAR(_npi_rcopysign_scalar)
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::Compute<cpu, mshadow_op::rcopysign>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_npi_rcopysign_scalar"});
+
+MXNET_OPERATOR_REGISTER_BINARY(_backward_npi_copysign_scalar)
+.add_argument("scalar", "float", "scalar value")
+.set_attr_parser([](NodeAttrs *attrs) { attrs->parsed = std::stod(attrs->dict["scalar"]); })
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::Backward<
+  cpu, mshadow_op::copysign_grad>);
```
Comment (haojin2) [inline src/operator/numpy/np_elemwise_broadcast_op.cc]: I would prefer:
```c++
.set_attr<FCompute>("FCompute<cpu>",
                    BinaryScalarOp::Backward<cpu, mshadow_op::copysign_grad>);
```
Same for the one below.

Comment (hzfan) [reply src/operator/numpy/np_elemwise_broadcast_op.cc]: fixed @haojin2

**Comment by hzfan** [comment]
PRed to the master branch. Closing this one.

**Pull request closed without merging**